cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/oauth/PutOAuthProviderRequest.java

```diff
@@ -26,22 +26,36 @@ public class PutOAuthProviderRequest {
   private final String clientId;
   private final String clientSecret;
   private final OAuthProvider.CredentialEncodingStrategy strategy;
+  private final String userAgent;
 
+  /** Constructor with default strategy & userAgent */
   public PutOAuthProviderRequest(String loginURL, String tokenRefreshURL, String clientId, String clientSecret) {
-    this(loginURL, tokenRefreshURL, clientId, clientSecret, OAuthProvider.CredentialEncodingStrategy.FORM_BODY);
+    this(loginURL, tokenRefreshURL, clientId, clientSecret, OAuthProvider.CredentialEncodingStrategy.FORM_BODY, null);
   }
 
+  /** Constructor with default userAgent (null) */
   public PutOAuthProviderRequest(
           String loginURL,
           String tokenRefreshURL,
           String clientId,
           String clientSecret,
           OAuthProvider.CredentialEncodingStrategy strategy) {
+    this(loginURL, tokenRefreshURL, clientId, clientSecret, strategy, null);
```

Comment:
can add `userAgent` in the same constructor instead of introducing a new one.

Actually, we don't even need separate constructor for `credentialEncodingStrategy` here. There can be only one constructor since it is not an API and only used in tests.